app/models/event.rb

```diff
@@ -19,15 +19,10 @@ def self.total_capital_gain(year)
   end
 
   def self.events_for_year(year)
-    # This is a bit ugly but at least we can still run
-    # both Postgres and SQLite
+    day = Date.new(year)
+    year_range = (day...day + 1.year)
 
-    day = Date.parse("1-1-#{year}")
-    beginning_of_year = day.at_beginning_of_year
-    end_of_year = day.at_end_of_year.to_date
-
-    where('executed_on >= ?', beginning_of_year)
-      .where('executed_on <= ?', end_of_year)
+    where(executed_on: year_range)
```

Comment:
Section 2.3.2 of http://guides.rubyonrails.org/active_record_querying.html#hash-conditions

--
Benjamin

> On 26/05/2015, at 16.30, Libo Cannici <notifications@github.com> wrote:
> 
> In app/models/event.rb:
> 
> >  
> > -    day = Date.parse("1-1-#{year}")
> > -    beginning_of_year = day.at_beginning_of_year
> > -    end_of_year = day.at_end_of_year.to_date
> > -
> > -    where('executed_on >= ?', beginning_of_year)
> > -      .where('executed_on <= ?', end_of_year)
> > +    where(executed_on: year_range)
> sure it works on Postgres?
> 
> —
> Reply to this email directly or view it on GitHub.
>